### PR TITLE
enhance: Add FlowgraphManager interface

### PR DESCRIPTION
**Pull request by XuanYang-cn**
- Change flowgraphManager to fgManagerImpl
- Change close to stop
- change execute to controlMemWaterLevel
- Change method name of fgManager for readability
- Add mockery for fgmanager

Issue: #28853

### CODE REVIEW

File: internal/datanode/data_node.go [hunk 1]
```diff
@@ -85,7 +85,7 @@ type DataNode struct {
 	cancel           context.CancelFunc
 	Role             string
 	stateCode        atomic.Value // commonpb.StateCode_Initializing
-	flowgraphManager *flowgraphManager
+	flowgraphManager *fgManagerImpl
```
Comment (congqixia) [inline internal/datanode/data_node.go:88]: why not use `FlowgraphManager` interface here?

File: internal/datanode/flow_graph_manager.go [hunk 2]
```diff
@@ -33,42 +33,69 @@ import (
 	"github.com/milvus-io/milvus/pkg/util/typeutil"
 )
 
-type flowgraphManager struct {
+type FlowgraphManager interface {
+	Start()
+	Stop()
+
+	// control the total memory water level by sync data.
+	controlMemWaterLevel(totalMemory uint64)
```
Comment (congqixia) [inline internal/datanode/flow_graph_manager.go:41]: shall we keep this private method in interface?

Comment (XuanYang-cn) [reply internal/datanode/flow_graph_manager.go:41]: True, forget to remove this

File: internal/datanode/data_node.go (hunk 1 above)
Comment (XuanYang-cn) [reply internal/datanode/data_node.go:88]: Ture, forgot to change it.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**